component/src/org/wso2/ballerina/connectors/oauth2/ClientConnector.bal

```diff
@@ -0,0 +1,173 @@
+package org.wso2.ballerina.connectors.oauth2;
+
+import ballerina.net.http;
+
+@Description { value:"OAuth2 client connector"}
+@Param { value:"baseUrl: The endpoint base url"}
+@Param { value:"accessToken: The access token of the account"}
+@Param { value:"clientId: The client Id of the account"}
+@Param { value:"clientSecret: The client secret of the account"}
+@Param { value:"refreshToken: The refresh token of the account"}
+@Param { value:"refreshTokenEP: The refresh token endpoint url"}
+public connector ClientConnector (string baseUrl, string accessToken, string clientId, string clientSecret,
+                                  string refreshToken, string refreshTokenEP, string refreshTokenPath) {
+
+    endpoint<http:HttpClient> httpConnectorEP {
+        create http:HttpClient(baseUrl, {});
+    }
+
+    string accessTokenValue;
+    http:HttpConnectorError e;
+
+    @Description { value:"Get with OAuth2 authentication"}
+    @Param { value:"path: The endpoint path"}
+    @Param { value:"request: The request of the method"}
+    @Return { value:"response object"}
+    @Return { value:"Error occured during HTTP client invocation." }
+    action get (string path, http:Request request) (http:Response, http:HttpConnectorError) {
+        http:Response response = {};
+
+        accessTokenValue = constructAuthHeader (request, accessTokenValue, accessToken);
```

Comment:
Why are you taking in "accessTokenValue" and also returning "accessTokenValue". No one is using the returned value. And you don't have to pass it in also, since it's a service level variable which is accessible by the "constructAuthHeader". Also, the name of the function is wrong, it should be something like "populateAuthHeader", since you're passing in the "request", and you're putting the header value inside the function.